src/solvers/solvers.cpp

```diff
@@ -13,23 +13,57 @@
 #include "eigenpy/solvers/LeastSquaresConjugateGradient.hpp"
 #endif
 
+#include "eigenpy/solvers/BiCGSTAB.hpp"
+#include "eigenpy/solvers/MINRES.hpp"
+
+#include "eigenpy/solvers/IncompleteLUT.hpp"
+#include "eigenpy/solvers/IncompleteCholesky.hpp"
+
 namespace eigenpy {
 void exposeSolvers() {
   using namespace Eigen;
-  ConjugateGradientVisitor<
-      ConjugateGradient<MatrixXd, Lower | Upper>>::expose();
+
+  using Eigen::Lower;
+
+  using Eigen::BiCGSTAB;
+  using Eigen::ConjugateGradient;
+  using Eigen::LeastSquaresConjugateGradient;
+
+  using Eigen::DiagonalPreconditioner;
+  using Eigen::IdentityPreconditioner;
+  using Eigen::LeastSquareDiagonalPreconditioner;
+
+  using IdentityBiCGSTAB = BiCGSTAB<MatrixXd, IdentityPreconditioner>;
+  using IdentityConjugateGradient =
+      ConjugateGradient<MatrixXd, Lower, IdentityPreconditioner>;
+  using IdentityLeastSquaresConjugateGradient =
+      LeastSquaresConjugateGradient<MatrixXd, IdentityPreconditioner>;
+  using DiagonalLeastSquaresConjugateGradient = LeastSquaresConjugateGradient<
+      MatrixXd, DiagonalPreconditioner<typename MatrixXd::Scalar>>;
+
+  ConjugateGradientVisitor<ConjugateGradient<MatrixXd, Lower>>::expose(
+      "ConjugateGradient");
+  ConjugateGradientVisitor<IdentityConjugateGradient>::expose(
+      "IdentityConjugateGradient");
+
 #if EIGEN_VERSION_AT_LEAST(3, 3, 5)
   LeastSquaresConjugateGradientVisitor<LeastSquaresConjugateGradient<
-      MatrixXd, LeastSquareDiagonalPreconditioner<MatrixXd::Scalar>>>::expose();
+      MatrixXd, LeastSquareDiagonalPreconditioner<MatrixXd::Scalar>>>::
+      expose("LeastSquaresConjugateGradient");
+  LeastSquaresConjugateGradientVisitor<IdentityLeastSquaresConjugateGradient>::
+      expose("IdentityLeastSquaresConjugateGradient");
+  LeastSquaresConjugateGradientVisitor<DiagonalLeastSquaresConjugateGradient>::
+      expose("DiagonalLeastSquaresConjugateGradient");
 #endif
 
-  // Conjugate gradient with limited BFGS preconditioner
-  ConjugateGradientVisitor<
-      ConjugateGradient<MatrixXd, Lower | Upper, IdentityPreconditioner>>::
-      expose("IdentityConjugateGradient");
-  //    ConjugateGradientVisitor<
-  //    ConjugateGradient<MatrixXd,Lower|Upper,LimitedBFGSPreconditioner<double,Dynamic,Lower|Upper>
-  //    > >::expose("LimitedBFGSConjugateGradient");
+  BiCGSTABVisitor<BiCGSTAB<MatrixXd>>::expose("BiCGSTAB");
+  BiCGSTABVisitor<IdentityBiCGSTAB>::expose("IdentityBiCGSTAB");
+
+  MINRESSolverVisitor<MatrixXd>::expose("MINRES");
```

Comment:
MINRES is now in the python eigenpy.solvers module. We must create an alias to still allow to call eigenpy.MINRES